### PR TITLE
update language on setting focus in a modal dialog

### PR DESCRIPTION
**Pull request by fstrr**
It seems like there is agreement in the discussion on issue #2241 on changing the language for setting focus in a modal dialog. This PR updates the Understanding document with that change.

### CODE REVIEW

**Review by patrickhlauke — APPROVED** [review]

**Comment by fstrr** [comment]
closes #2241

**Review by mbgower — COMMENTED** [review]
I wonder if "on or within" would be better? I'm pretty sure the html5 dialog now allows for focus to be placed in a variety of places, including on the dialog element itself. So "within" seems a bit too specific.

**Comment by scottaohara** [comment]
to be fair though, this example is saying it has implemented a modal dialog via scripting, so is this even an example of using the dialog element?

the default for a dialog in html is still the first focusable element within.  it's only specific author overrides / fallback scenarios that the dialog element itself can receive focus.  

tldr; yes, dialogs can have focus set on or within.  but in the context of the current text / high level examples presented in this file, does mentioning that even matter, as how exactly this dialog is implemented is still rather nebulous.

File: understanding/20/focus-order.html [hunk 1]
```diff
@@ -127,7 +127,7 @@ <h2>Examples of Focus Order</h2>
          </li>
          
          <li>A Web page implements modal dialogs via scripting. When the trigger button is activated,
-            a dialog opens and focus is set to the first interactive element in the dialog. As
+            a dialog opens and focus is set within the dialog. As
```
Comment (mbgower) [inline understanding/20/focus-order.html:130]:
```suggestion
            a dialog opens and focus is set on or within the dialog. As
```
